path_service_principal.go

```diff
@@ -122,15 +121,9 @@ func (b *azureSecretBackend) createSPSecret(ctx context.Context, s logical.Stora
 		return nil, err
 	}
 
-	// Pre-generate UUIDs to be provided to assignRoles so we can rollback if we need to
-	var assignmentIDs []string
-
-	for i := 0; i < len(role.AzureRoles); i++ {
-		assignmentID, err := uuid.GenerateUUID()
-		if err != nil {
-			return nil, err
-		}
-		assignmentIDs = append(assignmentIDs, assignmentID)
+	assignmentIDs, err := c.generateAssignmentIDsFromRole(role)
+	if err != nil {
+		return nil, err
```

Comment:
Same thing about the error here for Service Principal.

Reply:
Updated in latest commit